pkg/controller/directimagemigration/imagestream.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package directimagemigration
+
+import (
+	"context"
+	//  liberr "github.com/konveyor/controller/pkg/error"
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	imagev1 "github.com/openshift/api/image/v1"
+	kapi "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) getDirectImageStreamMigrations() ([]migapi.DirectImageStreamMigration, error) {
+	ismList := migapi.DirectImageStreamMigrationList{}
+	err := t.Client.List(
+		context.TODO(),
+		k8sclient.MatchingLabels(t.Owner.GetCorrelationLabels()),
+		&ismList)
+	if err != nil {
+		return nil, err
+	}
+	return ismList.Items, nil
+}
+
+func (t *Task) createDirectImageStreamMigrations() error {
+	// Get client for source
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+	// Get list namespaces to iterate over
+	for srcNsName, destNsName := range t.Owner.GetNamespaceMapping() {
+		isList := imagev1.ImageStreamList{}
+		err := srcClient.List(
+			context.TODO(),
+			k8sclient.InNamespace(srcNsName),
+			&isList)
+		if err != nil {
+			return err
+		}
+		for _, is := range isList.Items {
+			ismList := migapi.DirectImageStreamMigrationList{}
+			err := t.Client.List(
```

Comment:
I wonder if making this a Get call instead makes sense

Reply:
Doesn't the Get call require you to know the name? This is only used to make sure we don't create a new CR if one already exists for this `dim` and for this `is`, using label selectors to grab it. Because we need to use generateName (see above), we don't know the name.